src/path_converters.h

```diff
@@ -327,11 +316,30 @@ class PathClipper
 
     inline void rewind(unsigned path_id)
     {
-        m_has_next = false;
+        m_has_init = false;
         m_moveto = true;
         m_source->rewind(path_id);
     }
 
+    int draw_clipped_line(double x0, double y0, double x1, double y1)
+    {
+        unsigned moved = agg::clip_line_segment(&x0, &y0, &x1, &y1, m_cliprect);
+        // moved >= 4 - Fully clipped
+        // moved & 1 != 0 - First point has been moved
+        // moved & 2 != 0 - Second point has been moved
+        if (moved < 4) {
+            if (moved & 1 || m_moveto) {
+                queue_push(agg::path_cmd_move_to, x0, y0);
```

Comment:
We need some way to mark that this move is 'synthetic' so that the logic an line 366 knows not to re-set the m_initX / m_initY values.  I think that would allow us to clip polygons and fix #5948 more gracefully.